### PR TITLE
Added some details about Python GC

### LINKED ISSUE #345: Python garbage collection misleadingly described
From https://ocaml.org/learn/tutorials/garbage_collection.html:

> Python has a form of garbage collection, but it uses a simple scheme called reference counting. Simply put, each Python object keeps a count of the number of other objects pointing (referencing) itself. When the count falls to zero, nothing is pointing at this object, and so the object can be freed.
>
> Reference counting is not considered as serious garbage collection by computer scientists, yet it has one big practical advantage over full garbage collectors. [...]

Later, the page mentions the inability to collect cyclic data structures as a disadvantage of refcounting. That's true for refcounting as-is, but the reference to Python is misleading, since Python *does* collect cyclic data: it has *both* a refcounting scheme that reclaims most objects immediately after they get unused, *and* a mark-and-sweep collector that runs from time to time to collect cyclic structures.

It might work to simply replace the reference to Python with Swift, which only has reference counting as far as I can read (but I'm not expert in Swift).

### PR DESCRIPTION
**Pull request by cuihtlauac**
Fix: https://github.com/ocaml/ocaml.org/issues/345

### CODE REVIEW

**Review by tmattio — APPROVED** [review]
Thanks! I especially appreciate the removal of "serious garbage collector" in favour of "full fledge garbage collector".

File: data/tutorials/rt_00_garbage_collection.md [hunk 1]
```diff
@@ -47,16 +47,19 @@ The OCaml GC is synchronous. It doesn't run in a separate thread, and it
...
-Reference counting is not considered as serious garbage collection by
-computer scientists, yet it has one big practical advantage over full
-garbage collectors. With reference counting, you can avoid many explicit
-calls to `close`/`closedir` in code. Whereas in OCaml
+Python has a form of garbage collection, but it
+[mostly uses](https://devguide.python.org/internals/garbage-collector/) a
+simple scheme called
+[reference counting](https://en.wikipedia.org/wiki/Reference_counting).
+Simply put, each Python object keeps a count of the number of other
+objects pointing (referencing) itself. When the count falls to zero,
```
Comment (christinerose) [inline data/tutorials/rt_00_garbage_collection.md:55]: "objects pointing to (referencing) itself"
Missing the "to"

File: data/tutorials/rt_00_garbage_collection.md [hunk 2]
```diff
@@ -47,16 +47,19 @@ The OCaml GC is synchronous. It doesn't run in a separate thread, and it
...
+nothing is pointing at this object, and so the object can be freed.
```
Comment (christinerose) [inline data/tutorials/rt_00_garbage_collection.md:56]: "nothing is pointing at this object, so the object can be freed."
The "and" isn't needed here

File: data/tutorials/rt_00_garbage_collection.md [hunk 3]
```diff
@@ -47,16 +47,19 @@ The OCaml GC is synchronous. It doesn't run in a separate thread, and it
...
+
+Reference counting is automatic memory management but it is not considered
+a full fledged garbage collection by computer scientists, yet it has one
+big practical advantage over full garbage collectors. With reference counting,
+you can avoid many explicit calls to `close`/`closedir` in code.
+Whereas in OCaml
```
Comment (christinerose) [inline data/tutorials/rt_00_garbage_collection.md:62]: Reference counting is automatic memory management, but it is not considered
a full-fledged garbage collection by computer scientists. Yet it does have one
big practical advantage over full garbage collectors. With reference counting,
you can avoid many explicit calls to `close`/`closedir` in code.

(Some syntax/grammar stuff)
Whereas in OCaml

File: data/tutorials/rt_00_garbage_collection.md [hunk 4]
```diff
@@ -88,7 +91,9 @@ disadvantages of reference counting schemes:
  bloats the code size of compiled code.
 * They cannot collect so-called circular, or self-referential
  structures. I've programmed in many languages in many years and
- can't recall ever having created one of these.
+ can't recall ever having created one of these. In addition to
+ reference counting, Python has a dedicated garbage collector to
+ address such.
 * Graph algorithms, of course, violate the previous assumption.
```
Comment (christinerose) [inline data/tutorials/rt_00_garbage_collection.md:97]: They cannot collect so-called circular or self-referential
 structures. I've programmed in many languages over the years and
 can't recall ever having created one of these. In addition to
 reference counting, Python has a dedicated garbage collector to
 address this.

(Some syntax/grammar stuff)

**Review by christinerose — COMMENTED** [review]
I didn't see this until after merge, but I have some suggestions. Should I open a PR for them?

File: data/tutorials/rt_00_garbage_collection.md (hunk 1 above)
Comment (cuihtlauac) [reply data/tutorials/rt_00_garbage_collection.md:55]: Thanks @christinerose, sorry you saw things after the merge. I'll add your edits directly.

Comment (christinerose) [reply data/tutorials/rt_00_garbage_collection.md:55]: No worries! It's often the case since I'm on the other side of the world! :)

**Pull request merged**